File: apps/androidtv/core/designsystem/src/main/kotlin/dev/spidola/tv/core/designsystem/SpidolaSpacing.kt
// SPDX-FileCopyrightText: 2026 Spidola contributors
// SPDX-License-Identifier: AGPL-3.0-or-later

package dev.spidola.tv.core.designsystem

import androidx.compose.ui.unit.dp

/**
 * The spacing scale and TV-safe margins (PRD §8.4: "All content respects TV-safe margins").
 * The safe insets keep content off the ~5% overscan region every panel can clip.
 */
object SpidolaSpacing {
    val xs = 4.dp
    val s = 8.dp
    val m = 16.dp
    val l = 24.dp
    val xl = 48.dp

    /** Horizontal TV-safe inset (PRD §8.4). */
    val safeHorizontal = 48.dp

    /** Vertical TV-safe inset (PRD §8.4). */
    val safeVertical = 27.dp
}

File: apps/tvos/Packages/DesignSystem/Sources/DesignSystem/SpidolaSpacing.swift
// SPDX-FileCopyrightText: 2026 Spidola contributors
// SPDX-License-Identifier: AGPL-3.0-or-later

import CoreGraphics

/// The spacing scale and TV-safe insets (PRD §8.4: "All content respects TV-safe margins"). tvOS
/// supplies an overscan-safe area of its own; these are the content margins layered on top.
public enum SpidolaSpacing {
  public static let xs: CGFloat = 4
  public static let s: CGFloat = 8
  public static let m: CGFloat = 16
  public static let l: CGFloat = 24
  public static let xl: CGFloat = 48

  /// Horizontal content inset (PRD §8.4).
  public static let safeHorizontal: CGFloat = 48

  /// Vertical content inset (PRD §8.4).
  public static let safeVertical: CGFloat = 27
}
